c7n not deleting matched SG rule

### Describe the bug

Custodian policy was implemented to delete open ingress at the time of creation based on aws events. The policy was in place for more than 2 years and successfully deleting the violating SG rules


### What did you expect to happen?

Looking at cloud trail events, looks like the policy stopped working sometime in April. Cloud trail event shows custodian is attempting to delete the violating SG rule, but failing with the reason
" "errorCode": "Client.InvalidPermission.NotFound",
"errorMessage": "The specified rule does not exist in this security group.",


### Cloud Provider

Amazon Web Services (AWS)

### Cloud Custodian version and dependency information

```shell
Please copy/paste the following info along with any bug reports:

Custodian:   0.9.29
Python:      3.7.16 (default, May  6 2024, 19:30:00)
             [GCC 7.3.1 20180712 (Red Hat 7.3.1-17)]
Platform:    posix.uname_result(sysname='Linux', nodename='ip-10-35-225-59.ec2.internal', release='4.14.348-265.565.amzn2.x86_64', version='#1 SMP Fri Jun 28 23:44:17 UTC 2024', machine='x86_64')
Using venv:  False
Docker: False
Installed:

argcomplete==3.1.1
attrs==23.1.0
boto3==1.28.5
botocore==1.31.5
docutils==0.18.1
importlib-metadata==5.2.0
importlib-resources==5.12.0
jmespath==1.0.1
jsonschema==4.17.3
pkgutil-resolve-name==1.3.10
pyrsistent==0.19.3
python-dateutil==2.8.2
pyyaml==6.0.1
s3transfer==0.6.1
six==1.16.0
tabulate==0.9.0
typing-extensions==4.7.1
urllib3==1.26.16
zipp==3.15.0
```


### Policy

```shell
policies:
  - name: remove-violating-security-groups-ingress
    resource: security-group
    description: |
      Checks security groups for rules that allow unrestricted access to a resource.
    mode:
        type: cloudtrail
        role: arn:aws:iam::{account_id}:role/c7n-lambda
        events:
          - source: ec2.amazonaws.com
            event: AuthorizeSecurityGroupIngress
            ids: "requestParameters.groupId"
          - source: ec2.amazonaws.com
            event: ModifySecurityGroupRules
            ids: "requestParameters.ModifySecurityGroupRulesRequest.GroupId"
          - source: ec2.amazonaws.com
            event: AuthorizeSecurityGroupEgress
            ids: "requestParameters.groupId"
          - source: ec2.amazonaws.com
            event: RevokeSecurityGroupEgress
            ids: "requestParameters.groupId"
          - source: ec2.amazonaws.com
            event: RevokeSecurityGroupIngress
            ids: "requestParameters.groupId"
    filters:
      - "tag:maid_status": absent
      - type: ingress
        SelfReference: false
      - type: ingress
        OnlyPorts: [443]
        Cidr:
          value: "0.0.0.0/0"

    actions:
        - type: set-permissions
          remove-ingress: matched
```


### Relevant log/traceback output

```shell
INIT_START Runtime Version: python:3.9.v55	Runtime Version ARN: arn:aws:lambda:us-east-1::runtime:be9e7121d3264b1e86158b38dbbb656c23dff979eb481793ee37b9e2b79fda22
START RequestId: 4b5e0164-160f-451b-b8f9-823832751f96 Version: $LATEST
[DEBUG]	2024-08-02T16:16:33.190Z	4b5e0164-160f-451b-b8f9-823832751f96	using default region:us-east-1 from boto
[INFO]	2024-08-02T16:16:33.534Z	4b5e0164-160f-451b-b8f9-823832751f96	Processing event
 
{
    "version": "0",
    "id": "880a789a-f7f7-ef5d-2dec-1a08ecb8ccc1",
    "detail-type": "AWS API Call via CloudTrail",
    "source": "aws.ec2",
    "account": "XXXXXXXXXXX",
    "time": "2024-08-02T16:16:28Z",
    "region": "us-east-1",
    "resources": [],
    "detail": {
        "eventVersion": "1.09",
        "userIdentity": {
            "type": "AssumedRole",
            "principalId": "XXXXXXXXXXXXXXXXXXXXX:abc@xyz.com",
            "arn": "arn:aws:sts::XXXXXXXXXXX:assumed-role/AWS-XXXXXXXXXX-Admin/abc@xyz.com",
            "accountId": "XXXXXXXXXXX",
            "accessKeyId": "XXXXXXXXXXXXXXXXXXXXX",
            "sessionContext": {
                "sessionIssuer": {
                    "type": "Role",
                    "principalId": "XXXXXXXXXXXXXXXXXXXXX",
                    "arn": "arn:aws:iam::XXXXXXXXXXX:role/AWS-XXXXXXXXXX-Admin",
                    "accountId": "XXXXXXXXXXX",
                    "userName": "AWS-XXXXXXXXXX-Admin"
                },
                "attributes": {
                    "creationDate": "2024-08-02T14:52:45Z",
                    "mfaAuthenticated": "false"
                }
            }
        },
        "eventTime": "2024-08-02T16:16:28Z",
        "eventSource": "ec2.amazonaws.com",
        "eventName": "RevokeSecurityGroupIngress",
        "awsRegion": "us-east-1",
        "sourceIPAddress": "x.x.x.x",
        "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36",
        "requestParameters": {
            "groupId": "sg-0ad37a0fe82ed02a4",
            "ipPermissions": {
                "items": [
                    {
                        "ipProtocol": "-1",
                        "groups": {},
                        "ipRanges": {
                            "items": [
                                {
                                    "cidrIp": "0.0.0.0/0"
                                }
                            ]
                        },
                        "ipv6Ranges": {},
                        "prefixListIds": {}
                    }
                ]
            }
        },
        "responseElements": {
            "requestId": "5da5a212-23c0-4efe-abfd-3f3eda7bada1",
            "_return": true,
            "revokedSecurityGroupRuleSet": {
                "items": [
                    {
                        "groupId": "sg-0ad37a0fe82ed02a4",
                        "securityGroupRuleId": "sgr-0b63c288085e27082",
                        "isEgress": false,
                        "ipProtocol": "-1",
                        "fromPort": -1,
                        "toPort": -1,
                        "cidrIpv4": "0.0.0.0/0"
                    }
                ]
            }
        },
        "requestID": "5da5a212-23c0-4efe-abfd-3f3eda7bada1",
        "eventID": "bfef5726-14e0-4cd3-a589-b433086ef7d0",
        "readOnly": false,
        "eventType": "AwsApiCall",
        "managementEvent": true,
        "recipientAccountId": "XXXXXXXXXXX",
        "eventCategory": "Management",
        "tlsDetails": {
            "tlsVersion": "TLSv1.3",
            "cipherSuite": "TLS_AES_128_GCM_SHA256",
            "clientProvidedHostHeader": "ec2.us-east-1.amazonaws.com"
        },
        "sessionCredentialFromConsole": "true"
    },
    "debug": true
}

[DEBUG]	2024-08-02T16:16:33.534Z	4b5e0164-160f-451b-b8f9-823832751f96	Disabling cache
[WARNING]	2024-08-02T16:16:33.535Z	4b5e0164-160f-451b-b8f9-823832751f96	Custodian reserves policy lambda tags starting with custodian - policy specifies custodian-info
[INFO]	2024-08-02T16:16:33.535Z	4b5e0164-160f-451b-b8f9-823832751f96	Found resource ids:['sg-0ad37a0fe82ed02a4']
[INFO]	2024-08-02T16:16:34.251Z	4b5e0164-160f-451b-b8f9-823832751f96	Resources [{'Description': 'KM-Test-SG-01', 'GroupName': 'KM-Test-SG-01', 'IpPermissions': [], 'OwnerId': 'XXXXXXXXXXX', 'GroupId': 'sg-0ad37a0fe82ed02a4', 'IpPermissionsEgress': [], 'Tags': [{'Key': 'Service', 'Value': 'DevOps'}, {'Key': 'Contact', 'Value': 'karthik.mani'}, {'Key': 'CostCenter', 'Value': '11902438'}, {'Key': 'Capacity', 'Value': 'BAU'}, {'Key': 'OrgID', 'Value': '00491'}, {'Key': 'User', 'Value': 'karthik.mani'}, {'Key': 'environment', 'Value': 'dev'}, {'Key': 'appid', 'Value': '00491'}, {'Key': 'Patch Group', 'Value': 'Dev'}, {'Key': 'Company', 'Value': 'Osttra'}], 'VpcId': 'vpc-c61b18a3'}]
[INFO]	2024-08-02T16:16:34.251Z	4b5e0164-160f-451b-b8f9-823832751f96	Filtering resources using 3 filters
[DEBUG]	2024-08-02T16:16:34.252Z	4b5e0164-160f-451b-b8f9-823832751f96	Filter #1 applied 1->1 filter: 
{
    "tag:maid_status": "absent"
}

[DEBUG]	2024-08-02T16:16:34.252Z	4b5e0164-160f-451b-b8f9-823832751f96	Filter #2 applied 1->0 filter: 
{
    "type": "ingress",
    "SelfReference": false
}

[DEBUG]	2024-08-02T16:16:34.252Z	4b5e0164-160f-451b-b8f9-823832751f96	Filtered from 1 to 0 securitygroup
[INFO]	2024-08-02T16:16:34.252Z	4b5e0164-160f-451b-b8f9-823832751f96	Filtered resources 0 of 1
[INFO]	2024-08-02T16:16:34.252Z	4b5e0164-160f-451b-b8f9-823832751f96	policy:remove-violating-security-groups-ingress resources:security-group no resources matched
END RequestId: 4b5e0164-160f-451b-b8f9-823832751f96
REPORT RequestId: 4b5e0164-160f-451b-b8f9-823832751f96	Duration: 1592.68 ms	Billed Duration: 1593 ms	Memory Size: 512 MB	Max Memory Used: 108 MB	Init Duration: 581.71 ms	
START RequestId: 6f58fb6f-4958-4d21-bf72-c521004238dc Version: $LATEST
[INFO]	2024-08-02T16:16:48.722Z	6f58fb6f-4958-4d21-bf72-c521004238dc	Processing event
 
{
    "version": "0",
    "id": "142a5010-bbf4-eb6c-9dac-10d03e60717c",
    "detail-type": "AWS API Call via CloudTrail",
    "source": "aws.ec2",
    "account": "XXXXXXXXXXX",
    "time": "2024-08-02T16:16:45Z",
    "region": "us-east-1",
    "resources": [],
    "detail": {
        "eventVersion": "1.09",
        "userIdentity": {
            "type": "AssumedRole",
            "principalId": "XXXXXXXXXXXXXXXXXXXXX:abc@xyz.com",
            "arn": "arn:aws:sts::XXXXXXXXXXX:assumed-role/AWS-XXXXXXXXXX-Admin/abc@xyz.com",
            "accountId": "XXXXXXXXXXX",
            "accessKeyId": "XXXXXXXXXXXXXXXXXXXXX",
            "sessionContext": {
                "sessionIssuer": {
                    "type": "Role",
                    "principalId": "XXXXXXXXXXXXXXXXXXXXX",
                    "arn": "arn:aws:iam::XXXXXXXXXXX:role/AWS-XXXXXXXXXX-Admin",
                    "accountId": "XXXXXXXXXXX",
                    "userName": "AWS-XXXXXXXXXX-Admin"
                },
                "attributes": {
                    "creationDate": "2024-08-02T14:52:45Z",
                    "mfaAuthenticated": "false"
                }
            }
        },
        "eventTime": "2024-08-02T16:16:45Z",
        "eventSource": "ec2.amazonaws.com",
        "eventName": "AuthorizeSecurityGroupIngress",
        "awsRegion": "us-east-1",
        "sourceIPAddress": "x.x.x.x",
        "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36",
        "requestParameters": {
            "groupId": "sg-0ad37a0fe82ed02a4",
            "ipPermissions": {
                "items": [
                    {
                        "ipProtocol": "-1",
                        "groups": {},
                        "ipRanges": {
                            "items": [
                                {
                                    "cidrIp": "0.0.0.0/0"
                                }
                            ]
                        },
                        "ipv6Ranges": {},
                        "prefixListIds": {}
                    }
                ]
            }
        },
        "responseElements": {
            "requestId": "24e4d560-2753-4b57-a0c8-ebc6f5794e48",
            "_return": true,
            "securityGroupRuleSet": {
                "items": [
                    {
                        "groupOwnerId": "XXXXXXXXXXX",
                        "groupId": "sg-0ad37a0fe82ed02a4",
                        "securityGroupRuleId": "sgr-01b8a6f5470a61118",
                        "isEgress": false,
                        "ipProtocol": "-1",
                        "fromPort": -1,
                        "toPort": -1,
                        "cidrIpv4": "0.0.0.0/0"
                    }
                ]
            }
        },
        "requestID": "24e4d560-2753-4b57-a0c8-ebc6f5794e48",
        "eventID": "3b6def41-ce97-41ae-8eba-e9facda33527",
        "readOnly": false,
        "eventType": "AwsApiCall",
        "managementEvent": true,
        "recipientAccountId": "XXXXXXXXXXX",
        "eventCategory": "Management",
        "tlsDetails": {
            "tlsVersion": "TLSv1.3",
            "cipherSuite": "TLS_AES_128_GCM_SHA256",
            "clientProvidedHostHeader": "ec2.us-east-1.amazonaws.com"
        },
        "sessionCredentialFromConsole": "true"
    },
    "debug": true
}

[WARNING]	2024-08-02T16:16:48.723Z	6f58fb6f-4958-4d21-bf72-c521004238dc	Custodian reserves policy lambda tags starting with custodian - policy specifies custodian-info
[INFO]	2024-08-02T16:16:48.723Z	6f58fb6f-4958-4d21-bf72-c521004238dc	Found resource ids:['sg-0ad37a0fe82ed02a4']
[INFO]	2024-08-02T16:16:48.860Z	6f58fb6f-4958-4d21-bf72-c521004238dc	Resources [{'Description': 'KM-Test-SG-01', 'GroupName': 'KM-Test-SG-01', 'IpPermissions': [{'IpProtocol': '-1', 'IpRanges': [{'CidrIp': '0.0.0.0/0'}], 'Ipv6Ranges': [], 'PrefixListIds': [], 'UserIdGroupPairs': []}], 'OwnerId': 'XXXXXXXXXXX', 'GroupId': 'sg-0ad37a0fe82ed02a4', 'IpPermissionsEgress': [], 'Tags': [{'Key': 'Service', 'Value': 'DevOps'}, {'Key': 'Contact', 'Value': 'karthik.mani'}, {'Key': 'CostCenter', 'Value': '11902438'}, {'Key': 'Capacity', 'Value': 'BAU'}, {'Key': 'OrgID', 'Value': '00491'}, {'Key': 'User', 'Value': 'karthik.mani'}, {'Key': 'environment', 'Value': 'dev'}, {'Key': 'appid', 'Value': '00491'}, {'Key': 'Patch Group', 'Value': 'Dev'}, {'Key': 'Company', 'Value': 'Osttra'}], 'VpcId': 'vpc-c61b18a3'}]
[INFO]	2024-08-02T16:16:48.860Z	6f58fb6f-4958-4d21-bf72-c521004238dc	Filtering resources using 3 filters
[DEBUG]	2024-08-02T16:16:48.861Z	6f58fb6f-4958-4d21-bf72-c521004238dc	Filter #1 applied 1->1 filter: 
{
    "tag:maid_status": "absent"
}

[DEBUG]	2024-08-02T16:16:48.861Z	6f58fb6f-4958-4d21-bf72-c521004238dc	Filter #2 applied 1->1 filter: 
{
    "type": "ingress",
    "SelfReference": false
}

[DEBUG]	2024-08-02T16:16:48.861Z	6f58fb6f-4958-4d21-bf72-c521004238dc	Filter #3 applied 1->1 filter: 
{
    "type": "ingress",
    "OnlyPorts": [
        443
    ],
    "Cidr": {
        "value": "0.0.0.0/0",
        "key": "CidrIp"
    }
}

[DEBUG]	2024-08-02T16:16:48.861Z	6f58fb6f-4958-4d21-bf72-c521004238dc	Filtered from 1 to 1 securitygroup
[INFO]	2024-08-02T16:16:48.861Z	6f58fb6f-4958-4d21-bf72-c521004238dc	Filtered resources 1 of 1
[DEBUG]	2024-08-02T16:16:48.862Z	6f58fb6f-4958-4d21-bf72-c521004238dc	Storing output with <LogFile file:///tmp/remove-violating-security-groups-ingress/custodian-run.log>
[DEBUG]	2024-08-02T16:16:48.868Z	6f58fb6f-4958-4d21-bf72-c521004238dc	metric:ResourceCount Count:1 policy:remove-violating-security-groups-ingress restype:security-group scope:policy
[INFO]	2024-08-02T16:16:48.869Z	6f58fb6f-4958-4d21-bf72-c521004238dc	Invoking actions [<c7n.resources.vpc.SetPermissions object at 0x7f0fda16dca0>]
[INFO]	2024-08-02T16:16:48.869Z	6f58fb6f-4958-4d21-bf72-c521004238dc	policy:remove-violating-security-groups-ingress invoking action:setpermissions resources:1
[DEBUG]	2024-08-02T16:16:49.136Z	6f58fb6f-4958-4d21-bf72-c521004238dc	metric:PolicyException Count:1 policy:remove-violating-security-groups-ingress restype:security-group
[DEBUG]	2024-08-02T16:16:49.136Z	6f58fb6f-4958-4d21-bf72-c521004238dc	metric:ApiCalls Count:1 policy:remove-violating-security-groups-ingress restype:security-group
[ERROR]	2024-08-02T16:16:49.149Z	6f58fb6f-4958-4d21-bf72-c521004238dc	Error while executing policy
Traceback (most recent call last):
  File "/var/task/c7n/policy.py", line 553, in run_resource_set
    results = action.process(resources)
  File "/var/task/c7n/resources/vpc.py", line 1712, in process
    remover.process(resources)
  File "/var/task/c7n/resources/vpc.py", line 1606, in process
    method(GroupId=r['GroupId'], IpPermissions=groups)
  File "/var/runtime/botocore/client.py", line 553, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/var/runtime/botocore/client.py", line 1009, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (InvalidPermission.NotFound) when calling the RevokeSecurityGroupIngress operation: The specified rule does not exist in this security group.
[ERROR]	2024-08-02T16:16:49.153Z	6f58fb6f-4958-4d21-bf72-c521004238dc	error during policy execution
Traceback (most recent call last):
  File "/var/task/c7n/handler.py", line 167, in dispatch_event
    p.push(event, context)
  File "/var/task/c7n/policy.py", line 1317, in push
    return mode.run(event, lambda_ctx)
  File "/var/task/c7n/policy.py", line 519, in run
    return self.run_resource_set(event, resources)
  File "/var/task/c7n/policy.py", line 553, in run_resource_set
    results = action.process(resources)
  File "/var/task/c7n/resources/vpc.py", line 1712, in process
    remover.process(resources)
  File "/var/task/c7n/resources/vpc.py", line 1606, in process
    method(GroupId=r['GroupId'], IpPermissions=groups)
  File "/var/runtime/botocore/client.py", line 553, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/var/runtime/botocore/client.py", line 1009, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (InvalidPermission.NotFound) when calling the RevokeSecurityGroupIngress operation: The specified rule does not exist in this security group.
LAMBDA_WARNING: Unhandled exception. The most likely cause is an issue in the function code. However, in rare cases, a Lambda runtime update can cause unexpected function behavior. For functions using managed runtimes, runtime updates can be triggered by a function change, or can be applied automatically. To determine if the runtime has been updated, check the runtime version in the INIT_START log entry. If this error correlates with a change in the runtime version, you may be able to mitigate this error by temporarily rolling back to the previous runtime version. For more information, see https://docs.aws.amazon.com/lambda/latest/dg/runtimes-update.html
[ERROR] ClientError: An error occurred (InvalidPermission.NotFound) when calling the RevokeSecurityGroupIngress operation: The specified rule does not exist in this security group.
Traceback (most recent call last):
  File "/var/task/custodian_policy.py", line 4, in run
    return handler.dispatch_event(event, context)
  File "/var/task/c7n/handler.py", line 167, in dispatch_event
    p.push(event, context)
  File "/var/task/c7n/policy.py", line 1317, in push
    return mode.run(event, lambda_ctx)
  File "/var/task/c7n/policy.py", line 519, in run
    return self.run_resource_set(event, resources)
  File "/var/task/c7n/policy.py", line 553, in run_resource_set
    results = action.process(resources)
  File "/var/task/c7n/resources/vpc.py", line 1712, in process
    remover.process(resources)
  File "/var/task/c7n/resources/vpc.py", line 1606, in process
    method(GroupId=r['GroupId'], IpPermissions=groups)
  File "/var/runtime/botocore/client.py", line 553, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/var/runtime/botocore/client.py", line 1009, in _make_api_call
    raise error_class(parsed_response, operation_name)
END RequestId: 6f58fb6f-4958-4d21-bf72-c521004238dc
REPORT RequestId: 6f58fb6f-4958-4d21-bf72-c521004238dc	Duration: 469.25 ms	Billed Duration: 470 ms	Memory Size: 512 MB	Max Memory Used: 108 MB	
START RequestId: 4aaebe7e-3e7e-4b5c-87e2-24addfa4cd5b Version: $LATEST
[DEBUG]	2024-08-02T16:16:53.512Z	4aaebe7e-3e7e-4b5c-87e2-24addfa4cd5b	Skipping failed operation: Client.InvalidPermission.NotFound
END RequestId: 4aaebe7e-3e7e-4b5c-87e2-24addfa4cd5b
REPORT RequestId: 4aaebe7e-3e7e-4b5c-87e2-24addfa4cd5b	Duration: 1.47 ms	Billed Duration: 2 ms	Memory Size: 512 MB	Max Memory Used: 108 MB
```


### Extra information or context

{
"eventVersion": "1.09",
"userIdentity": {
"type": "AssumedRole",
"principalId": "XXXXXXXXXXXXXXXXXXXXXX:custodian-remove-violating-security-groups-ingress",
"arn": "arn:aws:sts::XXXXXXXXXXX:assumed-role/c7n-lambda/custodian-remove-violating-security-groups-ingress",
"accountId": "XXXXXXXXXXX",
"accessKeyId": "XXXXXXXXXXXXXXXXXXX",
"sessionContext": {
"sessionIssuer": {
"type": "Role",
"principalId": "XXXXXXXXXXXXXXXXXXXXXX",
"arn": "arn:aws:iam::XXXXXXXXXXX:role/c7n-lambda",
"accountId": "XXXXXXXXXXX",
"userName": "c7n-lambda"
},
"attributes": {
"creationDate": "2024-08-02T15:54:01Z",
"mfaAuthenticated": "false"
}
}
},
"eventTime": "2024-08-02T15:55:26Z",
"eventSource": "ec2.amazonaws.com",
"eventName": "RevokeSecurityGroupIngress",
"awsRegion": "us-east-1",
"sourceIPAddress": "x.x.x.x",
"userAgent": "c7n/0.9.29 md/Botocore#1.34.42 ua/2.0 os/linux#5.10.219-229.866.amzn2.x86_64 md/arch#x86_64 lang/python#3.9.19 md/pyimpl#CPython exec-env/AWS_Lambda_python3.9 cfg/retry-mode#legacy c7n/policy#remove-violating-security-groups-ingress",
"errorCode": "Client.InvalidPermission.NotFound",
"errorMessage": "The specified rule does not exist in this security group.",
"requestParameters": {
"groupId": "sg-0ad37a0fe82ed02a4",
"ipPermissions": {
"items": [
{
"ipProtocol": "-1",
"groups": {},
"ipRanges": {
"items": [
{
"cidrIp": "0.0.0.0/0"
}
]
},
"ipv6Ranges": {},
"prefixListIds": {}
},
{
"ipProtocol": "-1",
"groups": {},
"ipRanges": {
"items": [
{
"cidrIp": "0.0.0.0/0"
}
]
},
"ipv6Ranges": {},
"prefixListIds": {}
}
]
}
},
"responseElements": null,
"requestID": "363ffa57-c698-4323-9f6b-bb8992a967e9",
"eventID": "c1f149f8-23e7-4eec-a8d5-272d112b08d8",
"readOnly": false,
"eventType": "AwsApiCall",
"managementEvent": true,
"recipientAccountId": "XXXXXXXXXXX",
"eventCategory": "Management",
"tlsDetails": {
"tlsVersion": "TLSv1.3",
"cipherSuite": "TLS_AES_128_GCM_SHA256",
"clientProvidedHostHeader": "ec2.us-east-1.amazonaws.com"
}
}

--- a/c7n/resources/vpc.py
+++ b/c7n/resources/vpc.py
@@ -1614,7 +1614,11 @@ class SGPermission(Filter):
                 matched.append(perm)
 
         if matched:
-            resource.setdefault('Matched%s' % self.ip_permissions_key, []).extend(matched)
+            matched_annotation = resource.setdefault('Matched%s' % self.ip_permissions_key, [])
+            # If the same rule matches multiple filters, only add it to the match annotation
+            # once. Note: Because we're looking for unique dicts and those aren't hashable,
+            # we can't conveniently use set() to de-duplicate rules.
+            matched_annotation.extend(m for m in matched if m not in matched_annotation)
             return True
 
 

--- a/tests/test_vpc.py
+++ b/tests/test_vpc.py
@@ -3244,6 +3244,10 @@ class SecurityGroupTest(BaseTest):
                         "Cidr": {
                             "value": "10.42.1.239", "op": "in", "value_type": "cidr"
                         },
+                    },
+                    {
+                        "type": "ingress",
+                        "SelfReference": False
                     }
                 ],
             },

